src/main/java/Main.java

```diff
@@ -0,0 +1,75 @@
+import core.basesyntax.FruitTransaction;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import service.CsvReaderService;
+import service.CsvWriterService;
+
+public class Main {
+    public static void main(String[] args) {
+        CsvReaderService csvReaderService = new CsvReaderService();
+        CsvWriterService csvWriterService = new CsvWriterService();
+
+        String inputFilePath = "input.csv";
+        String outputFilePath = "output.csv";
+
+        try {
+            List<FruitTransaction> transactions = csvReaderService.readTransactions(inputFilePath);
+
+            Map<String, Integer> fruitInventory = new HashMap<>();
+
+            for (FruitTransaction transaction : transactions) {
+                String fruit = transaction.getFruit();
+                int quantity = transaction.getQuantity();
+
+                switch (transaction.getOperation()) {
+                    case BALANCE:
+                    case SUPPLY:
+                        fruitInventory.put(fruit, fruitInventory.getOrDefault(fruit, 0)
+                                + quantity);
+                        break;
+                    case PURCHASE:
+                        if (fruitInventory.containsKey(fruit)) {
+                            int currentQuantity = fruitInventory.get(fruit);
+                            if (currentQuantity >= quantity) {
+                                fruitInventory.put(fruit, currentQuantity - quantity);
+                            } else {
+                                throw new RuntimeException("Not enough " + fruit
+                                        + " in inventory");
+                            }
+                        } else {
+                            throw new RuntimeException("Fruit " + fruit
+                                    + " not found in inventory");
+                        }
+                        break;
+                    case RETURN:
+                        if (fruitInventory.containsKey(fruit)) {
+                            int currentQuantity = fruitInventory.get(fruit);
+                            fruitInventory.put(fruit, currentQuantity + quantity);
+                        } else {
+                            throw new RuntimeException("Fruit " + fruit
+                                    + " not found in inventory");
+                        }
+                        break;
+                    default:
+                        throw new RuntimeException("Unknown operation: "
+                                + transaction.getOperation());
+                }
```

Comment:
too messy code. 

using switch-case + if-else is very bad practice and hard to read. 
Think about making Strategy Pattern.